pkg/cmd/auth/status/status.go

```
@@ -139,6 +141,12 @@ func statusRun(opts *StatusOptions) error {
 		// not to since I wanted this command to be read-only.
 	}
 
+	if !isHostnameFound {
+		fmt.Fprintf(stderr,
+			"You are not logged into any GitHub hosts. Run %s to authenticate.\n", cs.Bold(fmt.Sprintf("gh auth login -h %s", opts.Hostname)))
```

We already print this message above: https://github.com/cli/cli/blob/4750e4ae18735375ef91e84fa5c148783743c379/pkg/cmd/auth/status/status.go#L71-L74

The message you added here isn't accurate for the condition we're testing. The issue you've set to fix is to warn that a hostname passed via the `-h` argument wasn't found. How about something like this?

```
"Hostname %q not found among authenticated GitHub hosts"
```

---

@mislav It seems that the code in `status_test.go` silently ignores the `stderr` message in favour of `wantErr`. 

https://github.com/cli/cli/blob/4750e4ae18735375ef91e84fa5c148783743c379/pkg/cmd/auth/status/status_test.go#L239-L243

Since we return `SilentError` we have to make `wantErr` non-nil, and that forces condition on L241 to occur, thus pre-maturely returning, avoiding a check on `wantErrOut`. Any  suggestions around this?